std/json.d

```diff
@@ -213,13 +213,21 @@ struct JSONValue
      * Value getter/setter for `JSONType.float_`. Note that despite
      * the name, this is a $(B 64)-bit `double`, not a 32-bit `float`.
      * Throws: `JSONException` for read access if `type` is not
-     * `JSONType.float_`.
+     * `JSONType.float_`, `JSONType.integer` or `JSONType.uinteger`.
      */
     @property double floating() const pure @safe
     {
-        enforce!JSONException(type == JSONType.float_,
-                                "JSONValue is not a floating type");
-        return store.floating;
+        switch(type)
+        {
+        case JSONType.float_:
+            return store.floating;
+        case JSONType.uinteger:
+            return cast(double)store.uinteger;
```

Comment:
I suggest removing this cast to emphasize that this is an implicit conversion according to D rules, per your argument above.